internal/buffer/settings.go

```diff
@@ -20,7 +20,19 @@ func (b *Buffer) SetOptionNative(option string, nativeValue interface{}) error {
 	} else if option == "statusline" {
 		screen.Redraw()
 	} else if option == "filetype" {
-		config.InitLocalSettings(b.Settings, b.Path)
+		parsedSettings := config.ParsedSettings()
+		parsedSettings[option] = nativeValue
```

Comment:
I confirm, that it looks hacky, but didn't find a better solution so far.
We need to start with clean parsed settings (local within this function) and call `InitLocalSettings()`, which then operates at this given map. But `filetype` must be set to parse the local settings (`filetype` dependent) correctly within this given map, otherwise it would ignore the `ft: XYZ` settings.
Since we don't want to clear it again to `unknown` or cause recursive calls it's ignored in the following loop.

Please correct me, in case I oversee something.

Reply:
> But `filetype` must be set to parse the local settings

Ah, indeed. Looks good then.

I'd maybe just use explicit `parsedSettings["filetype"]` instead of `parsedSettings[option]`, to make it a bit more clear.

Maybe also rename `parsedSettings` to just `settings`?